### DISCUSSION TITLE
Error with NordVPN

### DISCUSSION THREAD

**Discussion by yoann56100** [haugene/docker-transmission-openvpn#2203] [Q&A] [answered]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Docker run config used

version: "3"
services:
  vpn:

    image: haugene/transmission-openvpn
    container_name: OpenVPN
    environment:
      - OPENVPN_PROVIDER=NORDVPN
      - OPENVPN_CONFIG=france
      - OPENVPN_USERNAME=******
      - OPENVPN_PASSWORD=*****
      - LOCAL_NETWORK=192.168.100.0/16
      - NORDVPN_CATEGORY=legacy_p2p
      - NORDVPN_COUNTRY=FR
      - NORDVPN_PROTOCOL=udp

    logging:
      driver: json-file
      options:
        max-size: 10m
    volumes:
      - /home/noayoadmin/Desktop/transmission/data:/data

    ports:
      - 9091:9091
      - 51413:51413
      - 51413:51413/udp
      - 8112:8112
      - 6881:6881
      - 6881:6881/udp

### Current Behavior

Openvpn not connecting to my vpn

### Expected Behavior

Connect to the provider

### How have you tried to solve the problem?

1) i tried multiple times.
2)check my docker-compose config

### Log output

Starting container with revision: 8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
Downloading user specified config. NORDVPN_PROTOCOL is set to: udp
2022-03-02 20:48:13 Checking curl installation
2022-03-02 20:48:13 Removing existing configs
2022-03-02 20:48:13 Selecting the best server...
2022-03-02 20:48:13 Searching for country : FR (74)
2022-03-02 20:48:13 Searching for technology: openvpn_udp
2022-03-02 20:48:13 Best server : fr611.nordvpn.com
2022-03-02 20:48:13 Downloading config: default.ovpn
2022-03-02 20:48:13 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/fr611.nordvpn.com.udp.ovpn
2022-03-02 20:48:13 Using OpenVPN CONFIG :: france
2022-03-02 20:48:13 Downloading config: france.ovpn
2022-03-02 20:48:13 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/france.udp.ovpn
Starting OpenVPN using config france.ovpn
Modifying /etc/openvpn/nordvpn/france.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 192.168.100.0/16 via 172.21.0.1 dev eth0
RTNETLINK answers: Operation not permitted
Options error: Unrecognized option or missing or extra parameter(s) in /etc/openvpn/nordvpn/france.ovpn:1: html (2.4.7)
Use --help for more information.

### HW/SW Environment

```markdown
- OS:Debian 10.10 on Vmware Esxi
- Docker:20.10.12 (API: 1.41)
```

**Comment by edgd1er** [comment]
At least for Nordvpn, for the moment, use the dev branch, as the script has been modified a lot.

The error is due to an html file used as an openvpn config file, this happens when the container restarts too often and hits a nordvpn throttle limit. 
though, I would have expected an error message after this line:
https://github.com/haugene/docker-transmission-openvpn/blob/85d3b9633607f2d1fd58d21190a84a31a4cea7b8/openvpn/nordvpn/configure-openvpn.sh#L259

I suspect the root cause is the openvpn_config=france
Could you restart the container without that value (france), the country is read from NORDVPN_COUNTRY if set.

https://haugene.github.io/docker-transmission-openvpn/provider-specific/

**Reply by yoann56100** [reply]
i removed the openvpn_config=france, but now i have a different error:

Starting container with revision: 8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
2022-03-03 17:20:16 Checking curl installation
2022-03-03 17:20:16 Removing existing configs
2022-03-03 17:20:16 Selecting the best server...
2022-03-03 17:20:16 Searching for technology: openvpn_udp
2022-03-03 17:20:16 Best server : fr688.nordvpn.com
2022-03-03 17:20:16 Downloading config: default.ovpn
2022-03-03 17:20:16 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/fr688.nordvpn.com.udp.ovpn
2022-03-03 17:20:16 Selecting the best server...
2022-03-03 17:20:16 Searching for country : FR (74)
2022-03-03 17:20:16 Searching for technology: openvpn_udp
2022-03-03 17:20:16 Best server : fr688.nordvpn.com
2022-03-03 17:20:16 Downloading config: fr688.nordvpn.com.ovpn
2022-03-03 17:20:16 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/fr688.nordvpn.com.udp.ovpn
Starting OpenVPN using config fr688.nordvpn.com.ovpn
Modifying /etc/openvpn/nordvpn/fr688.nordvpn.com.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 192.168.100.0/16 via 172.21.0.1 dev eth0
RTNETLINK answers: Operation not permitted
Thu Mar  3 17:20:18 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021
Thu Mar  3 17:20:18 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Thu Mar  3 17:20:18 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Thu Mar  3 17:20:18 2022 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Thu Mar  3 17:20:18 2022 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Thu Mar  3 17:20:18 2022 TCP/UDP: Preserving recently used remote address: [AF_INET]45.152.181.51:1194
Thu Mar  3 17:20:18 2022 Socket Buffers: R=[212992->212992] S=[212992->212992]
Thu Mar  3 17:20:18 2022 UDP link local: (not bound)
Thu Mar  3 17:20:18 2022 UDP link remote: [AF_INET]45.152.181.51:1194
Thu Mar  3 17:20:18 2022 TLS: Initial packet from [AF_INET]45.152.181.51:1194, sid=05b3056c ee8aceb2
Thu Mar  3 17:20:18 2022 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Thu Mar  3 17:20:18 2022 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
Thu Mar  3 17:20:18 2022 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA7
Thu Mar  3 17:20:18 2022 VERIFY KU OK
Thu Mar  3 17:20:18 2022 Validating certificate extended key usage
Thu Mar  3 17:20:18 2022 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Thu Mar  3 17:20:18 2022 VERIFY EKU OK
Thu Mar  3 17:20:18 2022 VERIFY OK: depth=0, CN=fr688.nordvpn.com
Thu Mar  3 17:20:18 2022 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 4096 bit RSA
Thu Mar  3 17:20:18 2022 [fr688.nordvpn.com] Peer Connection Initiated with [AF_INET]45.152.181.51:1194
Thu Mar  3 17:20:19 2022 SENT CONTROL [fr688.nordvpn.com]: 'PUSH_REQUEST' (status=1)
Thu Mar  3 17:20:20 2022 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,sndbuf 524288,rcvbuf 524288,explicit-exit-notify,comp-lzo no,route-gateway 10.8.1.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.8.1.8 255.255.255.0,peer-id 6,cipher AES-256-GCM'
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: timers and/or timeouts modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: explicit notify parm(s) modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: compression parms modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified
Thu Mar  3 17:20:20 2022 Socket Buffers: R=[212992->425984] S=[212992->425984]
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: --ifconfig/up options modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: route options modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: route-related options modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: peer-id set
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: adjusting link_mtu to 1657
Thu Mar  3 17:20:20 2022 OPTIONS IMPORT: data channel crypto options modified
Thu Mar  3 17:20:20 2022 Data Channel: using negotiated cipher 'AES-256-GCM'
Thu Mar  3 17:20:20 2022 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Thu Mar  3 17:20:20 2022 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Thu Mar  3 17:20:20 2022 ROUTE_GATEWAY 172.21.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:15:00:02
Thu Mar  3 17:20:20 2022 ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1)
Thu Mar  3 17:20:20 2022 Exiting due to fatal error

**Reply by yoann56100** [reply]
okay, i acheive to fix by adding "privileged: true" but i cant acces to the port that i specify in the docker-compose

**Comment by edgd1er** [comment]
privilieged: true is really the last option I would use, did you try to add /dev/tun to devices, net_admin capabilities ?

In the container, only 9091 is a listening port. I'm not sure Nordvpn has a port forwarding feature: https://nordvpn.com/fr/blog/port-forwarding/

transmission web site should be available on 9091 depending if you have or not these lines:
```
transmission  | STARTING TRANSMISSION
transmission  | Transmission startup script complete.
```

all other ports are useless:
```
 -  51413:51413
  - 51413:51413/udp
  - 8112:8112
  - 6881:6881
  - 6881:6881/udp
```

**Reply by yoann56100** [reply]
the other ports are not useless, im using deluge (cause i prefer deluge to transmission) but none of the ports in the list (at least the ports for webUI) is accesible...

version: "3"
services:
  vpn:

    image: haugene/transmission-openvpn
    container_name: OpenVPN
    privileged: true
    environment:
      - OPENVPN_PROVIDER=NORDVPN
#      - OPENVPN_CONFIG=france
      - OPENVPN_USERNAME=********
      - OPENVPN_PASSWORD=********
      - LOCAL_NETWORK=192.168.100.0/16
      - NORDVPN_CATEGORY=legacy_p2p
      - NORDVPN_COUNTRY=FR
      - NORDVPN_PROTOCOL=udp
    logging:
      driver: json-file
      options:
        max-size: 10m
    volumes:
      - /home/noayoadmin/Desktop/transmission/data:/data

    ports:
      - 9091:9091
      - 51413:51413
      - 51413:51413/udp
      - 8112:8112
      - 6881:6881
      - 6881:6881/udp

  torrent:
    image: lscr.io/linuxserver/transmission
    container_name: transmission
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Berlin
      - TRANSMISSION_WEB_HOME=/combustion-release/ #optional
      - USER=******* #optional
      - PASS=******* #optional
#      - WHITELIST=iplist #optional
#      - PEERPORT=peerport #optional
      - HOST_WHITELIST=dnsnane list #optional
      - TRANSMISSION_UMASK=002
    volumes:
      - /home/noayoadmin/Desktop/transmission/config:/config
      - /home/noayoadmin/Desktop/Download:/downloads
      #- <path to watch folder>:/watch

    network_mode: service:vpn
    depends_on:
      - vpn
    restart: unless-stopped

  deluge:
    image: lscr.io/linuxserver/deluge
    container_name: deluge

**Comment by edgd1er** [comment, marked as answer]
well , this is clearly out of the scope of this project. 9091 is the port for transmission. You wish to use another bittorrent client, no problem, but i'm afraid this is not the place to seek support for other bittorent client than transmission and/or docker network configuration.
giving bits of configuration here and there will not help neither. mixing  the same folders for multiple transmission clients. 
Should you bittorrent client starts, look at the logs.

**Reply by yoann56100** [reply]
everything start fine, even the transmission client, but i cant acces it with the port 9091

Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.8.0.9
Updating Transmission settings.json with values from env variables
Generating settings.json for Transmission from environment and defaults /etc/transmission/default-settings.json
Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.8.0.9
Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed
Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete
Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch
sed'ing True to true

Transmission will run as
User name:   root
User uid:    0
User gid:    0
STARTING TRANSMISSION
Transmission startup script complete.

**Reply by pkishino** [reply]
Your container starts as expected, most likely your Local _network is wrong, please check the documentation
